Automattic/wp-calypso
# composite checkout: fix null contact fields

#### Changes proposed in this Pull Request

If an account has no cached contact details, or for some reason has `null` contact details cached, the cached details pre-fill introduced in #39997 may cause problems. For example, the country selector may seem to have a selection, but when it tries to validate, it will display an error.

![Screen Shot 2020-03-19 at 8 21 58 PM](https://user-images.githubusercontent.com/2036909/77126393-4dcc8180-6a1f-11ea-9e00-b237a80cf03b.png)

This happens because a `null` value makes the form fields uncontrolled. It should be impossible because the fields are managed with typescript that verifies that they are always strings. However, the cached data comes from `composite-checkout.js`, which is not typed.

This PR changes the assumption that the data from the cache is always strings and instead allows it to be `null`. It then changes null values into empty strings when setting the value.

**Before:**

![Screen Shot 2020-03-19 at 8 19 54 PM](https://user-images.githubusercontent.com/2036909/77126343-24135a80-6a1f-11ea-96ab-7d782bf795ab.png)

**After:**

![Screen Shot 2020-03-19 at 8 20 39 PM](https://user-images.githubusercontent.com/2036909/77126355-2ecdef80-6a1f-11ea-9fdd-f7998c24a642.png)


#### Testing instructions

With a new account (or one that has never made a purchase), try visiting composite checkout with just a plan in the cart and click through to the second step.

Make sure the postal code field is empty and the country code _does not_ have a country selected.

Fill in both fields and click "Continue" and verify that you are able to continue to the next step.
